hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -52,7 +52,29 @@ def unlink(self):
     @api.multi
     def _check_state_on_write(self, values):
         """ Hook for extensions """
-        self._check_state()
+        if not self._timesheet_validate_write(values):
+            self._check_state()
+
+    @api.model
+    def _timesheet_validate_write(self, values):
+        """ Hook for extensions """
+        return all([field_name not in values for field_name in
```

Comment:
Isn't simpler, more readable and more efficient to put?:
 ```python
return bool(set(self._get_timesheet_protected_fields()) - set(values.keys()))
```
and invert also the other `if` in `_check_state_on_write`.

Reply:
My intent was to allow to extend checks using more complex logic like "if only both fields are set, then fail"

Reply:
Yeah, and it's not incompatible. This is only a rewrite of the current code in a more efficient way and even more readable I think.

Reply:
But I'm afraid the expression is not correct. It should be an intersection of sets: `set(...).intersection(set(...)). Maybe not so readable right now, but simpler and more efficient.

Reply:
`set(...) & set(...)` also works as intersections

Reply:
@pedrobaeza agreed, though I'd leave `not` & `not` otherwise it should be `_timesheet_*in*validate_write`?

Reply:
Well, I interpret the contrary reading the method name: `_timesheet_validate_write` name suggest me that if return `True`, then it's going to validate (to check), that's to say, to perform the validation/constraint. Maybe renaming to `_should_check_timesheet_write` is more explanatory.

Reply:
Sounds good!